lib/matplotlib/gridspec.py

```diff
@@ -599,7 +599,10 @@ def _from_subplot_args(figure, args):
                 )
             i, j = num
         else:
-            if not isinstance(num, Integral) or num < 1 or num > rows*cols:
```

Comment:
I would slightly prefer making the previous error message more complete, e.g. "num must be an integer with 1 <= num <= ..., not ...", instead of delivering the info in a piecemeal fashion.  (If I accidentally pass 0.5 this will help me fixing the entire error (both type and value) at once, rather than first slapping an int() call, re-running the code, and seeing the second error.)